src/sink/to-array.ts

```diff
@@ -0,0 +1,22 @@
+// sink/to-array.ts
+
+/*
+ * Copyright (c) 2021 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import asyncerator, { Asyncable } from '../asyncerator';
+
+/**
+ * Turn an async iterable iterator into an Array.
+ * This will wait until the iterator is done before returning an array, so be careful using this
+ * with endless iterators (in other words, don't do that).
```

Comment:
Maybe enfore some upper limit for iterators so, endless iterators are not used?

Reply:
we need the concept of endless iterators/streams, e.g. a socket connection